app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,89 @@
+class OrdersController < ApplicationController
+
+  skip_before_action :require_login, except: [:index]
+
+  def index
+    if params[:merchant_id]
+      unless allowed_user(params[:merchant_id])
+        return
+      end
+      @merchant = Merchant.find_by(id: params[:merchant_id])
+      if params[:status_id] == "paid"
+        @order_items = @merchant.order_items_by_status("paid")
+      elsif params[:status_id] == "pending"
+        @order_items = @merchant.order_items_by_status("pending")
+      else
+        params[:status_id] == "all"
+        @order_items = @merchant.order_items
+      end
+      render :merchant_orders
+    end
+  end
+
+  def create
+    @order = Order.new(status: "pending")
+    @order.save
+    redirect_to orders_path
+  end
+
+  def update
+    #binding.pry
+    @order = Order.find(params[:id])
+
+    @order.customer_name = params[:order][:customer_name]
+    @order.customer_email = params[:order][:customer_email]
+    @order.customer_address = params[:order][:customer_address]
+    @order.cc_number = params[:order][:cc_number]
+    @order.cc_expiration = params[:order][:cc_expiration]
+    @order.cc_ccv = params[:order][:cc_ccv]
+    @order.zip_code = params[:order][:zip_code]
+
+    params[:order].each do |key, value|
+      if value == ""
+        flash[:status] = :failure
+        flash[:message] = "Any of required fields can't be empty"
+        redirect_to edit_order_path(@order)
+        return
+      end
```

Comment:
Instead of doing a `redirect_to` here you should use `render`, so that data the user has entered will stay in the form to get fixed.